app/cli/main.go

```diff
@@ -16,25 +16,86 @@
 package main
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"os"
+	"strings"
 
 	"github.com/chainloop-dev/chainloop/app/cli/cmd"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/posthog"
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1"
 	"github.com/go-kratos/kratos/v2/errors"
 	jwtMiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
 	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc/status"
 )
 
+// Follow the convention stated on https://consoledonottrack.com/
+const doNotTrackEnv = "DO_NOT_TRACK"
+
+var (
+	// Posthog API key and endpoint to be injected in the build process
+	posthogAPIKey   = ""
+	posthogEndpoint = ""
+)
+
 func main() {
 	// Couldn't find an easier way to disable the timestamp
 	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr, FormatTimestamp: func(interface{}) string { return "" }})
 	rootCmd := cmd.NewRootCmd(logger)
 
-	if err := rootCmd.Execute(); err != nil {
-		msg, exitCode := errorInfo(err, logger)
+	// Set the exit code, in case of an error it will be updated
+	var exitCode int
+
+	// Run the command and get the executed command
+	executedCmd, err := rootCmd.ExecuteC()
+	if err != nil {
+		var msg string
+		msg, exitCode = errorInfo(err, logger)
 		logger.Error().Msg(msg)
-		os.Exit(exitCode)
+	}
+
+	if !isTelemetryDisabled() {
+		recordCommand(logger, executedCmd, exitCode)
+	}
+
+	os.Exit(exitCode)
+}
+
+// recordCommand sends the command to the telemetry service
+func recordCommand(logger zerolog.Logger, executedCmd *cobra.Command, exitCode int) {
+	telemetryClient, err := posthog.NewClient(posthogAPIKey, posthogEndpoint)
+	if err != nil {
+		logger.Debug().Err(err).Msgf("creating telemetry client: %v", err)
```

Comment:
this debug entry seems to be shown even if debug level is not enabled. 

```
./bin/chainloop wf ls
ERR authentication required, please run "chainloop auth login"
DBG creating telemetry client: invalid configuration, API Key and endpoint URL are required error="invalid configuration, API Key and endpoint URL are required"
```

not sure why, or if this is the expected behavior.

Reply:
ok, the problem is that this logger is not correctly initialized the one in root is

```
func initLogger(logger zerolog.Logger) (zerolog.Logger, error) {
	lvl := zerolog.InfoLevel
	if flagDebug {
		lvl = zerolog.DebugLevel
	}

	return logger.Level(lvl), nil
}
```

This means that the logger I believe is in debug mode out of the box, not great I think. Not sure how to solve it though. At least we should make sure that these errors that are not meant to scare the user are not shown.

Reply:
I can see that a way to do this would be to newRootCommand to return also the initialized logger, to main can use it accordingly

Reply:
not trivial, we could leave it for now but create a task to propagate the level to main, wdyt?

Reply:
Yeah...I was giving a try this morning precisely with what you mention of returning the logger but it seems it does not take into account the previously set level...
I'll create a task for it.

Reply:
well,not really, do you also see this message on every command?

```
./bin/chainloop wf ls                                                  
DBG Telemetry enabled, to disable it use DO_NOT_TRACK=1
```

Reply:
Yes in all commands, it does not matter the value of the debug flag it will always be shown.